File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        buyAt = prices[0]
        profit = 0
        for price in range(1, len(prices)):
            if buyAt > prices[price]:
                buyAt = prices[price]
            profit = max(profit, prices[price] - buyAt)
        return profit

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        //asume buy at zero
        int buyAt = prices[0];
        int maxProfit = Integer.MIN_VALUE;
        for(int index = 0; index < prices.length; index++){
            if(buyAt > prices[index]){
                buyAt = prices[index];
            }
            // System.out.println(buyAt + " " + prices[index]);
            maxProfit = Math.max(maxProfit,  prices[index] - buyAt);
        }
        return maxProfit;
    }
}